core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -64,8 +65,8 @@ public Table createTable(
       baseLocation = defaultWarehouseLocation(identifier);
     }
 
-    TableMetadata metadata = TableMetadata.newTableMetadata(
-        schema, spec, baseLocation, properties == null ? Maps.newHashMap() : properties);
+    Map<String, String> tableProps = properties != null ? properties : ImmutableMap.of();
```

Comment:
Should we have defaults for `sortOrder` and `spec` as well? It seems strange that we do this for properties, but not the others.

Reply:
(Not something we need to decide in this PR)

Reply:
Yeah, I think that will be safe. Right now, we pass null for table properties in overloaded methods so we must handle it explicitly. But nothing prohibits a user from passing null for `spec` and `sortOrder` even though we properly default them in overloaded methods.

Reply:
I've created #1411.